File: src/test/resources/KThToLast.java
import com.jalgoarena.type.ListNode;

public class Solution {
    class Index {
        public int value = -1;
    }

    /**
     * @param head Linked List where we need to find kth to last element
     * @param k index of element to find
     * @return  kth to last element of a singly linked list
     */
    public int kthToLast(ListNode head, int k) {
        Index idx = new Index();
        return kthToLast(head, k, idx).value;
    }

    private ListNode kthToLast(ListNode head, int k, Index idx) {
        if (head == null) {
            return new ListNode(0);
        }

        ListNode node = kthToLast(head.next, k, idx);
        idx.value += 1;

        if (idx.value == k) {
            return head;
        }

        return node;
    }
}

File: src/test/resources/KThToLast.kt
import com.jalgoarena.type.ListNode
import com.jalgoarena.type.*

class KThToLast {
    inner class Index {
        var value = -1
    }

    /**
     * @param head Linked List where we need to find kth to last element
     * *
     * @param k index of element to find
     * *
     * @return  kth to last element of a singly linked list
     */
    fun kthToLast(head: ListNode, k: Int): Int {
        val idx = Index()
        return kthToLastPassingIndex(head, k, idx).value
    }

    private fun kthToLastPassingIndex(head: ListNode?, k: Int, idx: Index): ListNode {
        if (head == null) {
            return ListNode(0)
        }

        val node = kthToLastPassingIndex(head.next, k, idx)
        idx.value += 1

        if (idx.value == k) {
            return head
        }

        return node
    }
}
